x-pack/plugins/enterprise_search/public/applications/shared/layout/nav.tsx

```diff
@@ -38,12 +39,18 @@ import { KibanaLogic } from '../kibana';
 
 import { generateNavLink } from './nav_link_helpers';
 
-export const useEnterpriseSearchNav = () => {
+/**
+ * Hook to generate the Enterprise Search navigation items
+ *
+ * @param alwaysReturn Flag to always return the nav items, even if the sidebar is disabled
+ * @returns The Enterprise Search navigation items
+ */
+export const useEnterpriseSearchNav = (alwaysReturn = false) => {
```

Comment:
Adding the `alwaysReturn` is the best way I found to make the least amount of changes in the code base consuming this hook.

Reply:
we can actually remove the `isSidebarEnabled` code if we're disabling the solution nav in the templates.

`isSidebarEnabled` was added when we were expecting to include `enterprise_search` plugin in Serverless and we needed a global value to never populate the solution nav.

Reply:
This hook is used in a large number of place (and tests). To avoid increasing the scope and the LOC changed in this PR I would prefer that your team do a follow up to remove the `isSidebarEnabled` code if you don't think it is useful anymore.

<img width="730" alt="Screenshot 2024-05-20 at 11 51 47" src="https://github.com/elastic/kibana/assets/2854616/dadf5bc7-10a7-4e48-91e1-721631cb184a">

Reply:
I'm fine with that, I'm planning work related to support project nav for the legacy apps, so I'll scope in this refactor with that work.